File: 6_kyu/Character_with_longest_consecutive_repetition.py
from itertools import groupby
from typing import Tuple

def longest_repetition(chars: str) -> Tuple[str,int]:
    "Character with longest consecutive repetition"
    if not (chars and type(chars) is str):
        return ('',0)
    return max(((c,len(list(g))) for c,g in groupby(chars)), key=lambda x: x[1])

File: 6_kyu/Character_with_longest_consecutive_repetition.lua
local solution = {}

-- Search longest consecutive repetition of a chacacter.
-- Argument:
--    test - String to scan
-- Return 
--    Object {character=x,length=x} with data of longest repetition.
function solution.longest_repetition(text)
  local last = ""
  local last_len = 0
  local c = ""
  local len = 0
  for i = 1,#text do
    local now = text:sub(i,i)
    if last == now then
      last_len = last_len + 1
    else
      if last_len > len then
        len = last_len
        c = last
      end
      last = now
      last_len = 1
    end
  end
  if last_len > len then
    len = last_len
    c = last
  end
  return {character=c,length=len}
end

return solution
